DscResources/xPhpProvision/xPhpProvision.Schema.psm1

```diff
@@ -1,134 +1,143 @@
-            DestinationPath = $phpZip
-        }
-
-        # Make sure the content of the PHP archine are in the PHP path
-        Archive php
-        {
-            Path = $phpZip
-            Destination  = $DestinationPath
-        }
-
-        if ($installMySqlExt )
-        {               
-            # Make sure the MySql extention for PHP is in the main PHP path
-            File phpMySqlExt
-            {
-                SourcePath = "$($DestinationPath)\ext\php_mysql.dll"
-                DestinationPath = "$($DestinationPath)\php_mysql.dll"
-                Ensure = "Present"
-                DependsOn = @("[Archive]PHP")
-                MatchSource = $true
-            }
-        }
-
-            
-            # Make sure the php.ini is in the Php folder
-            File PhpIni
-            {
-                SourcePath = $ConfigurationPath
-                DestinationPath = "$($DestinationPath)\php.ini"
-                DependsOn = @("[Archive]PHP")
-                MatchSource = $true
-            }
-
-
-            # Make sure the php cgi module is registered with IIS
-            xIisModule phpHandler
-            {
-               Name = "phpFastCgi"
-               Path = "$($DestinationPath)\php-cgi.exe"
-               RequestPath = "*.php"
-               Verb = "*"
-               Ensure = "Present"
-               DependsOn = @("[Package]vcRedist","[File]PhpIni") 
-
-               # Removed because this dependency does not work in Windows Server 2012 R2 and below
-               # This should work in WMF v5 and above
-               # "[IisPreReqs_php]Iis" 
-            }
-
-        # Make sure the php binary folder is in the path
-        Environment PathPhp
-        {
-            Name = "Path"
-            Value = ";$($DestinationPath)"
-            Ensure = "Present"
-            Path = $true
-            DependsOn = "[Archive]PHP"
-        }
-}
-
-
+# Composite configuration to install the IIS pre-requisites for php
+Configuration IisPreReqs_php
+{
+param
+    (
+        [Parameter(Mandatory = $true)]
+        [Validateset("Present","Absent")]
+        [String]
+        $Ensure
+    )    
+
+    foreach ($Feature in @("Web-Server","Web-Mgmt-Tools","web-Default-Doc","Web-Dir-Browsing","Web-Http-Errors","Web-Static-Content",`
+            "Web-Http-Logging","web-Stat-Compression","web-Filtering",`
+            "web-CGI","web-ISAPI-Ext","web-ISAPI-Filter"))
+    {
+        WindowsFeature "$Feature$Number"
+        {
+            Ensure = $Ensure
+            Name = $Feature
+        }
+    }
+}
+
+# Composite configuration to install PHP on IIS
+configuration xPhpProvision
+{
+    param(
+        [Parameter(Mandatory = $true)]
+        [switch] $installMySqlExt,
+
+        [Parameter(Mandatory = $true)]
+        [string] $PackageFolder,
+
+        [Parameter(Mandatory = $true)]
+        [string] $DownloadUri,
+
+        [Parameter(Mandatory = $true)]
+        [string] $Vc2012RedistDownloadUri,

```

Comment:
The urls change frequently, so they should stay data.
Updated the destination parameters to have defaults.